### ISSUE TITLE
No parameter checking with non-direct extensions of classes with __init_subclass__ 

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#1704]
**Describe the bug**
If class `A` defines `__init_subclass__` with custom parameters, then the definition of a child class `B` gets checked against the arguments of `__init_subclass__` as expected. But when another class, `C`, extends `B`, its definition parameters are not checked against `A.__init_subclass__`. 

**To Reproduce**
Run the following through pyright:

```python
# pyright: strict

class A:
    def __init_subclass__(cls, param_a: int):
        super().__init_subclass__()
        print(f"Registering class {cls} with param {param_a}")

class B(A, param_a=123):
    pass

class C(B): # this is a runtime error, but it produces no pyright errors 
    pass
```

**Expected behavior**
An error on the declaration of class `C`, accusing the missing argument for `param_a`.

Mypy fails with `Missing positional argument "param_a" in call to "__init_subclass__" of "A"`

**VS Code extension or command-line**
Pyright extension version 1.1.123 and command line version 1.1.126

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.127, which I just published. It will also be included in the next release of pylance.
